src/docdb/tree/DocDBTreeItemBase.ts

```diff
@@ -43,22 +43,16 @@ export abstract class DocDBTreeItemBase<T> extends AzureParentTreeItem<IDocDBTre
         const resources: T[] = [];
         let count: number = 0;
         while (count < this._batchSize) {
-            const resource: T | undefined = await new Promise<T | undefined>((resolve, reject) => {
-                // tslint:disable-next-line: no-non-null-assertion
-                this._iterator!.nextItem((error: QueryError, rsrc: T | undefined) => {
-                    error ? reject(error) : resolve(rsrc);
-                });
-            });
-            if (resource === undefined) {
+            const resourceArray: T[] | undefined = (await this._iterator.fetchNext()).resources;
+            if (resourceArray === undefined) {
                 this._hasMoreChildren = false;
                 break;
             } else {
-                resources.push(resource);
+                resources.push(...resourceArray);
```

Comment:
If it's an array now instead of a single object, we'll have to re-do how we calculate batchSize

Reply:
I checked this method again after we talked. The current version actually works, except that with the new sdk we need set the `maxItemCount = 1` so that `fetchNext()` behaves the same way as `nextItem()`.

Reply:
I'm not sure what you mean by "actually works". If `_batchSize` is 10, my theory is that it's actually loading 10 batches (aka 100 items) every time a user clicks load more. To fix, I would get rid of the while loop so that it only loads one batch - this should be simpler and more efficient than changing `maxItemCount` to 1